File: BE-Task_1-Go/internal/repository/interfaces/refund_repository.go
package interfaces

import (
	"context"

	"gotest/internal/domain/entities"
)

type RefundRepository interface {
	Create(ctx context.Context, refund entities.Refund) error
	FindByID(ctx context.Context, refundID string) (entities.Refund, bool, error)
	ListAll(ctx context.Context) ([]entities.Refund, error)
}

File: BE-Task_1-Php/internal/repository/interfaces/refund_repository.php
<?php

declare(strict_types=1);

namespace App\Internal\Repository\Interfaces;

use App\Internal\Domain\Entities\Refund;

interface RefundRepositoryInterface
{
    public function create(Refund $refund): void;

    public function findById(string $refundId): ?Refund;

    /**
     * @return list<Refund>
     */
    public function listAll(): array;
}
